homeassistant/components/climate/dyson.py

```diff
@@ -0,0 +1,194 @@
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if self._device.environmental_state:
+            temperature_kelvin = self._device.environmental_state.temperature
+            if temperature_kelvin != 0:
+                self._current_temp = float("{0:.1f}".format(
+                    temperature_kelvin - 273))
+        return self._current_temp
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        if self._target_temp is None:
+            heat_target = int(self._device.state.heat_target) / 10
+            return int(heat_target - 273)
+        return self._target_temp
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        if self._device.environmental_state:
+            if self._device.environmental_state.humidity == 0:
+                return None
+            return self._device.environmental_state.humidity
+        return None
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        from libpurecoollink.const import HeatMode, HeatState
+        if self._device.state.heat_mode == HeatMode.HEAT_ON:
+            if self._device.state.heat_state == HeatState.HEAT_STATE_ON:
+                return STATE_HEAT
+            return STATE_IDLE
+        return STATE_COOL
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return [STATE_HEAT, STATE_COOL]
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        from libpurecoollink.const import FocusMode
+        if self._device.state.focus_mode == FocusMode.FOCUS_ON:
+            return STATE_FOCUS
+        return STATE_DIFFUSE
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        return [STATE_FOCUS, STATE_DIFFUSE]
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        target_temp = kwargs.get(ATTR_TEMPERATURE)
+        if target_temp is None:
+            return
+        target_temp = int(target_temp)
+        _LOGGER.debug("Set %s temperature %s", self.name, target_temp)
+        # Limit the target temperature into acceptable range.
+        target_temp = min(self.max_temp, target_temp)
+        target_temp = max(self.min_temp, target_temp)
+        self._target_temp = target_temp
+        # Dyson only response when it is in heat mode.
+        from libpurecoollink.const import HeatMode
+        if self._device.state.heat_mode == HeatMode.HEAT_ON:
+            from libpurecoollink.const import HeatTarget
+            if self.temperature_unit == TEMP_CELSIUS:
+                self._device.set_configuration(
+                    heat_target=HeatTarget.celsius(self._target_temp))
+            elif self.temperature_unit == TEMP_FAHRENHEIT:
```

Comment:
If we only need to use Celsius when interfacing with the device, we don't need this anymore.

Is the device showing the temperature outside home assistant to the user, and that's why we need both unit systems?